File: rosbook_ws/src/ch11_cpp_examples/src/talker.cpp
#include <chrono>
#include <memory>
#include <string>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Talker : public rclcpp::Node
{
public:
  Talker() : Node("ch11_cpp_talker"), count_(0)
  {
    publisher_ = create_publisher<std_msgs::msg::String>("ch11_chatter", 10);
    timer_ = create_wall_timer(std::chrono::milliseconds(500), [this]() {
      std_msgs::msg::String message;
      message.data = "Hello ROS 2 from C++: " + std::to_string(++count_);
      publisher_->publish(message);
    });
  }

private:
  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  int count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<Talker>());
  rclcpp::shutdown();
  return 0;
}

File: rosbook_ws/src/ch11_python_examples/ch11_python_examples/talker.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String


class Talker(Node):
    def __init__(self):
        super().__init__('ch11_python_talker')
        self.publisher = self.create_publisher(String, 'ch11_chatter', 10)
        self.count = 0
        self.timer = self.create_timer(0.5, self.publish_message)

    def publish_message(self):
        self.count += 1
        self.publisher.publish(String(data=f'Hello ROS 2: {self.count}'))


def main(args=None):
    rclpy.init(args=args)
    node = Talker()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()
